### PR TITLE
[CM-2348]: Implement equivalent of API.get_registry_model_details

### PR DESCRIPTION
**Pull request by yaricom**
### comet-java-client
- Implemented `CometApi.getRegistryModelDetails()` allowing to get registry model overview from Comet.
- Implemented `CometApi.getRegistryModelVersion()` allowing to get details about particular version of the registry model from Comet.

### comet-examples
- Updated `RegistryModelExample` to include example of retrieving model's general overview as well as details about particular version of the model.

### CODE REVIEW

File: comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelDetailsResponse.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    private String registryModelId;
+    private String workspaceId;
+    private String modelName;
+    private String description;
+    private long numberOfVersions;
+    @JsonProperty("isPublic")
+    private boolean isPublic;
+    private boolean isAutoGenerated;
+    private boolean isMonitored;
+    private List<RegistryModelItemDetails> versions;
```
[the remark this answers is not in the document]
Comment (yaricom) [reply comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelDetailsResponse.java:32]: @sonatype-lift ignore

File: comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelDetailsResponse.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    private String imagePath;
```
[the remark this answers is not in the document]
Comment (yaricom) [reply comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelDetailsResponse.java:33]: @sonatype-lift ignore

File: comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelDetailsResponse.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+public class RegistryModelDetailsResponse {
+    private String registryModelId;
+    private String workspaceId;
+    private String modelName;
```
[the remark this answers is not in the document]
Comment (yaricom) [reply comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelDetailsResponse.java:25]: @sonatype-lift ignore

File: comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelDetailsResponse.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+public class RegistryModelDetailsResponse {
+    private String registryModelId;
```
[the remark this answers is not in the document]
Comment (yaricom) [reply comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelDetailsResponse.java:23]: @sonatype-lift ignore

File: comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelDetailsResponse.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    private String modelName;
+    private String description;
+    private long numberOfVersions;
+    @JsonProperty("isPublic")
+    private boolean isPublic;
+    private boolean isAutoGenerated;
+    private boolean isMonitored;
+    private List<RegistryModelItemDetails> versions;
+    private String imagePath;
+    private String userName;
```
[the remark this answers is not in the document]
Comment (yaricom) [reply comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelDetailsResponse.java:34]: @sonatype-lift ignore

File: comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelDetailsResponse.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+import java.util.List;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+public class RegistryModelDetailsResponse {
+    private String registryModelId;
+    private String workspaceId;
```
[the remark this answers is not in the document]
Comment (yaricom) [reply comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelDetailsResponse.java:24]: @sonatype-lift ignore

File: comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelDetailsResponse.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+import org.slf4j.Logger;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@NoArgsConstructor
+@AllArgsConstructor
```
[the remark this answers is not in the document]
Comment (yaricom) [reply comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelDetailsResponse.java:20]: @sonatype-lift ignore

File: comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelDetailsResponse.java [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+@Data
```
[the remark this answers is not in the document]
Comment (yaricom) [reply comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelDetailsResponse.java:21]: @sonatype-lift ignore

File: comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelItemDetails.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class RegistryModelItemDetails {
+    private String registryModelItemId;
```
[the remark this answers is not in the document]
Comment (yaricom) [reply comet-java-client/src/main/java/ml/comet/experiment/impl/rest/RegistryModelItemDetails.java:22]: @sonatype-lift ignore

**Review by Lothiraldan — APPROVED** [review]
LGTM

**Pull request merged**